basics/trey-hunner-zip-and-enumerate.md

```diff
@@ -75,63 +75,138 @@ b 2
 c 3
 >>>
 ```
+[comment]: # (Zip and Enumerarate by Tushar starts here)
 
-Now you're ready to read [this awesome looping
-tutorial](http://treyhunner.com/2016/04/how-to-loop-with-indexes-in-python/).
-Read it now, then come back here and do the exercises.
-
-## Exercises
-
-1. Create a program that works like this. Here I entered everything
-    after the `>` prompt that the program displayed.
-
-    ```
-    Enter something, and press Enter without typing anything when you're done.
-    >hello there
-    >this is a test
-    >it seems to work
-    >
-    Line 1 is: hello there
-    Line 2 is: this is a test
-    Line 3 is: it seems to work
-    ```
-
-2. Create a program that prints all letters from A to Z and a to z
-    next to each other:
-
-    ```
-    A a
-    B b
-    C c
-    ...
-    X x
-    Y y
-    Z z
-    ```
-
-    Start your program like this:
-
-    ```python
-    uppercase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
-    lowercase = 'abcdefghijklmnopqrstuvwxyz'
-    ```
-
-    **Hint:** how do strings behave with `zip`? Try it out on the
-    `>>>` prompt and see.
-
-3. Can you make it print the indexes also?
-
-    ```
-    1 A a
-    2 B b
-    3 C c
-    ...
-    24 X x
-    25 Y y
-    26 Z z
-    ```
-
-The answers are [here](answers.md).
+## zip
+
+What comes to your mind when you hear the word `zip`? A mechanism extensively used to tie two parts of something i.e. Shirt, Jacket etc. The `zip()` functions does pretty much same, it helps us tie two or more different items together.
+
+### Syntax
+
+`zip(iterator1, iterator2, iterator3 ...) `
+
+### Parameter Values
+
+`zip()` can take 2 or more iterator values like so `zip(iterator1, iterator2, iterator3....)`. These
+iterators will be joined together once the function is executed.
+
+### Example 1
+
+Let's look at an example, suppose we wish to join 2 tuples together.
```

Comment:
I think all of this (from `### Syntax` to `Let's look at an example, ...`) can be deleted:
- The example below shows how it works. It is better to first show how it works, and then explain it as simply as you can if needed. So any explanation before showing is just unnecessary, and more likely to be confusing than useful: beginners don't yet have the skill to ignore the noise and scroll down to the code.
- Taking more than 2 iterables is a detail that IMO doesn't belong in a tutorial about Python basics.
- The `### Example 1` is unnecessary. Just show the example. It is obviously the first example in the file, no need to emphasize that.